### PR TITLE
Recognize .mjs as JavaScript files

### PR DESCRIPTION
**Pull request by mathiasbynens**
This is a quick and dirty fix for https://github.com/Polymer/tools/issues/736.

### CODE REVIEW

File: packages/analyzer/src/core/analysis-context.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export class AnalysisContext {
   readonly parsers = new Map<string, Parser<ParsedDocument>>([
     ['html', new HtmlParser()],
     ['js', new JavaScriptParser()],
+    ['mjs', new JavaScriptParser()],
```
Comment (mathiasbynens) [inline packages/analyzer/src/core/analysis-context.ts]: Of course, you could create a single `const jsParser = new JavaScriptParser();` and then use the the same instance for both `js` and `mjs`.

Comment (mathiasbynens) [reply packages/analyzer/src/core/analysis-context.ts]: You can also skip all the `import`/`export` scanning for `.mjs` files.

**Comment by mathiasbynens** [comment]
What's blocking this? @bicknellr @justinfagnani @rictic @aomarks @usergenic

File: packages/analyzer/src/core/analysis-context.ts (hunk 1 above)
Comment (rictic) [reply packages/analyzer/src/core/analysis-context.ts]: Yeah, we want to add more contextual inference for treating a file as a script vs module, and .mjs would be a strong signal.

No objection to landing this as is though.

**Pull request merged**